dcp_prototype/backend/ledger/code/common/ledger_orm.py

```diff
@@ -24,161 +24,232 @@ def session(self, **kwargs):
-    )
-    sequencing_protocol = relationship(
-        "SequencingProtocol", back_populates="sequence_files"
-    )
+    id = Column(String, primary_key=True)
+    filename = Column(String)
+    file_format = Column(String)
+    file_size = Column(String)
+    flowcell_id = Column(String)
+    lane_index = Column(String)
+    read_index = Column(String)
+    s3_uri = Column(String, nullable=False)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
 
 
 class SequencingProtocol(Base):
     __tablename__ = "sequencing_protocol"
-    id = Column(String(), primary_key=True)
-    paired_end_sequencing = Column(String(), nullable=True)
-    instrument_manufacturer_model = Column(String(), nullable=True)
-    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
-    updated_at = Column(
-        DateTime, default=datetime.utcnow, nullable=False, onupdate=datetime.utcnow
+
+    id = Column(String, primary_key=True)
+    paired_end_sequencing = Column(String)
+    instrument_manufacturer_model = Column(String)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+
+
+class AnalysisFileQuantificationProtocolExpressionFileProcessJ(Base):
+    __tablename__ = "analysis_file_quantification_protocol_expression_file_process_j"
+
+    id = Column(String, primary_key=True)
+    analysis_file_id = Column(ForeignKey("analysis_file.id"), nullable=False)
+    quantification_protocol_id = Column(
+        ForeignKey("quantification_protocol.id"), nullable=False
     )
+    expression_file_id = Column(ForeignKey("expression_file.id"), nullable=False)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+
+    analysis_file = relationship("AnalysisFile")
+    expression_file = relationship("ExpressionFile")
+    quantification_protocol = relationship("QuantificationProtocol")
+
+
+class BamQcMetric(Base):
+    __tablename__ = "bam_qc_metric"
+
+    id = Column(String, primary_key=True)
+    analysis_file_id = Column(ForeignKey("analysis_file.id"), nullable=False)
+    percent_mapped_to_genome = Column(String)
+    percent_reads_mapped_to_intergenic = Column(String)
+    percent_reads_mapped_uniquely = Column(String)
+    percent_reads_mapped_multiple = Column(String)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+
+    analysis_file = relationship("AnalysisFile")
+
+
+class Library(Base):
+    __tablename__ = "library"
+
+    id = Column(String, primary_key=True)
+    project_id = Column(ForeignKey("project.id"), nullable=False)
```

Comment:
Huh? this only contains a project_id.

Reply:
That is all the library entity contains in the schema. I don't understand its point but have told thats the way it is represented with potentially additional fields coming later